src/Sil/Component/Product/Repository/AttributeTypeRepositoryInterface.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * Copyright (C) 2015-2018 Libre Informatique
+ *
+ * This file is licenced under the GNU LGPL v3.
+ * For the full copyright and license information, please view the LICENSE.md
+ * file that was distributed with this source code.
+ */
+
+namespace Sil\Component\Product\Repository;
+
+use Blast\Component\Resource\Repository\ResourceRepositoryInterface;
+use Sil\Component\Product\Model\AttributeType;
+
+interface AttributeTypeRepositoryInterface extends ResourceRepositoryInterface
+{
+    /**
+     * Retreive attribute type by its name.
+     *
+     * @param string $name The name of the attribute type
+     *
+     * @return AttributeType|null
+     */
+    public function getAttributeTypeByName(string $name): ?AttributeType;
```

Comment:
`get*` method cannot return `null`, it throw an exception if the expected object does not exist.
Use `find*` method instead if you want to return `Something|null`.

Reply:
OK I'll use find methods instead